modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/ModalAddDefaultSortColumn/ModalAddDefaultSortColumn.tsx

```diff
@@ -0,0 +1,182 @@
+
+	useEffect(() => {
+		const newAvailableViewColumns = objectViewColumns.filter(
+			(viewColumn) => viewColumn.isDefaultSort === false
+		);
+
+		setAvailableViewColumns(newAvailableViewColumns);
+	}, [objectViewColumns]);
+
+	const sortOptions = [
+		{
+			label: Liferay.Language.get('ascending'),
+			value: 'asc',
+		},
+		{
+			label: Liferay.Language.get('descending'),
+			value: 'desc',
+		},
+	];
+
+	const [selectedObjectSortColumn, setSelectedObjectSortColumn] = useState<
+		TObjectViewSortColumn
+	>();
+	const [selectedObjetSort, setSelectedObjetSort] = useState(sortOptions[0]);
+	const [query, setQuery] = useState<string>('');
+
+	const filtredObjectSortColumn = useMemo(() => {
+		return availableViewColumns.filter(({objectFieldName}) => {
+			return objectFieldName.toLowerCase().includes(query.toLowerCase());
+		});
+	}, [availableViewColumns, query]);
+
+	const onSubmit = (event: FormEvent) => {
+		event.preventDefault();
+
+		const objectFieldName = selectedObjectSortColumn?.objectFieldName;
+
+		isEditingSort
+			? dispatch({
+					payload: {
+						editingObjectFieldName,
+						selectedObjectSort: selectedObjetSort.value,
+					},
+					type: TYPES.EDIT_OBJECT_VIEW_SORT_COLUMN_SORT_ORDER,
+			  })
+			: dispatch({
+					payload: {
+						objectFieldName,
+						objectFields,
+						objectViewSortColumns,
+						selectedObjetSort,
+					},
+					type: TYPES.ADD_OBJECT_VIEW_SORT_COLUMN,
+			  });
+
+		onClose();
+	};
+
+	return (
+		<ClayModal observer={observer}>
+			<ClayForm onSubmit={(event) => onSubmit(event)}>
+				<ClayModal.Header>{header}</ClayModal.Header>
+
+				<ClayModal.Body>
+					{!isEditingSort && (
+						<AutoComplete
+							emptyStateMessage={Liferay.Language.get(
+								'there-are-no-columns-added-in-this-view-yet'
+							)}
+							items={filtredObjectSortColumn}
+							label={Liferay.Language.get('columns')}
+							onChangeQuery={setQuery}
+							onSelectItem={(item) => {
+								setSelectedObjectSortColumn(item);
+							}}
+							query={query}
+							required
+							value={selectedObjectSortColumn?.label}
+						>
+							{({label}) => (
+								<div className="d-flex justify-content-between">
+									<div>{label}</div>
+								</div>
+							)}
+						</AutoComplete>
+					)}
+
+					<CustomSelect
+						label={Liferay.Language.get('sorting')}
+						onChange={(item: any) => {
+							setSelectedObjetSort(item);
+						}}
+						options={sortOptions}
+						value={selectedObjetSort.label}
+					>
+						{({label}) => (
+							<>
+								<div>{label}</div>
+							</>
+						)}
+					</CustomSelect>
+				</ClayModal.Body>
+
+				<ClayModal.Footer
+					last={
+						<ClayButton.Group key={1} spaced>
+							<ClayButton
+								displayType="secondary"
+								onClick={() => onClose()}
+							>
+								{Liferay.Language.get('cancel')}
+							</ClayButton>
+
+							<ClayButton
+								disabled={
+									isEditingSort
+										? false
+										: !selectedObjectSortColumn
```

Comment:
Please update this logic to be like

`!isEditingSort || !selectedObjectSortColumn` test if it is ok